spec/jobs/reports/sp_active_users_report_spec.rb

```diff
@@ -13,30 +12,72 @@
   end
 
   it 'returns total active user counts per sp broken down by ial1 and ial2' do
-    now = Time.zone.now
+    job_date = Date.new(2022, 1, 1)
+    authenticated_time = job_date.noon
     ServiceProvider.create(issuer: issuer, friendly_name: issuer, app_id: 'app')
     ServiceProviderIdentity.create(
       user_id: 1, service_provider: issuer, uuid: 'foo1',
-      last_ial1_authenticated_at: now, last_ial2_authenticated_at: now
+      last_ial1_authenticated_at: authenticated_time, last_ial2_authenticated_at: authenticated_time
     )
     ServiceProviderIdentity.create(
       user_id: 2, service_provider: issuer, uuid: 'foo2',
-      last_ial1_authenticated_at: now
+      last_ial1_authenticated_at: authenticated_time
     )
     ServiceProviderIdentity.create(
       user_id: 3, service_provider: issuer, uuid: 'foo3',
-      last_ial2_authenticated_at: now
+      last_ial2_authenticated_at: authenticated_time
     )
     ServiceProviderIdentity.create(
       user_id: 4, service_provider: issuer, uuid: 'foo4',
-      last_ial2_authenticated_at: now
+      last_ial2_authenticated_at: authenticated_time
     )
     result = [{ issuer: issuer, app_id: app_id, total_ial1_active: 2,
                 total_ial2_active: 3 }].to_json
 
     expect(subject.perform(Time.zone.today)).to eq(result)
   end
 
+  it 'when Oct 1, returns total active user counts per sp by ial1 and ial2 for last fiscal year' do
+    # run date is from 2019-10-01 to 2020-09-30
+    job_date = Date.new(2020, 10, 1)
+    beginning_of_last_fiscal_year = job_date.change(year: 2019, month: 10, day: 1).beginning_of_day
+    end_of_last_fiscal_year = job_date.change(year: 2020, month: 9, day: 30).end_of_day
+    middle_of_last_fiscal_year = job_date.change(year: 2020, month: 1, day: 31).end_of_day
+
+    beginning_of_current_fiscal_year = job_date.beginning_of_day
```

Comment:
this seems like a hint to me that maybe we could add a method to the class that we could expose for easy + clear testing:

```ruby
# @return [Range<Time>]
def reporting_ramge(date)
  # ....
  start_time(date)..finish_time(date)
end
```

then we could have a test assertion like the comment here

Reply:
added in 888ed9998 along with specs for it and the three cases (before Oct 1, on Oct 1, and after Oct 1)